compose/runtime/runtime/src/nonEmulatorCommonTest/kotlin/androidx/compose/runtime/CompositionTests.kt

```diff
@@ -3760,49 +3763,50 @@ class CompositionTests {
         }
     }
 
-    @Test(timeout = 10000)
-    fun testCompositionAndRecomposerDeadlock() {
-        runBlocking {
-            withGlobalSnapshotManager {
-                repeat(100) {
-                    val job = Job(parent = coroutineContext[Job])
-                    val coroutineContext = Dispatchers.Unconfined + job
-                    val recomposer = Recomposer(coroutineContext)
-
-                    launch(
-                        coroutineContext + BroadcastFrameClock(),
-                        start = CoroutineStart.UNDISPATCHED
-                    ) {
-                        recomposer.runRecomposeAndApplyChanges()
-                    }
+    @Test
+    fun testCompositionAndRecomposerDeadlock() = runTest(
```

Comment:
Can we write it on the next line to avoid merge conflicts?

```
fun testCompositionAndRecomposerDeadlock() {
    runTest(
        timeoutMs = 10_000,
        context = UnconfinedTestDispatcher()
    ) {
```

Reply:
In order to make this test work in k/js, runTest result should be retruned by the test function, so I'll change it like this:

```kotlin
    @Test
    fun testCompositionAndRecomposerDeadlock(): TestResult {
        return runTest(
            timeoutMs = 10_000,
            context = UnconfinedTestDispatcher()
        ) {
        ```

Reply:
From the docs of `runTest`:

> On JVM and Native, this function behaves similarly to runBlocking, with the difference that the code that it runs will skip delays. This allows to use delay in without causing the tests to take more time than necessary. On JS, this function creates a Promise that executes the test body with the delay-skipping behavior.

Reply:
Final code:

```kotlin
    @Test
    fun testCompositionAndRecomposerDeadlock(): TestResult {
        return runTest(timeoutMs = 10_000, context = UnconfinedTestDispatcher()) {
```